mlir/tools/mlir-miopen-lib/CMakeLists.txt

```diff
@@ -125,45 +126,90 @@ if( BUILD_FAT_LIBMLIRMIOPEN )
     # Install lib${target_name}.a to ${CMAKE_INSTALL_PREFIX}/lib/<package>/
     # and group it into the ${export-set}
     # The INTERFACE_INCLUDE_DIRECTORIES will be set as the property of the exported library
-    # so that when comsumer libraries (MIOpen) link in ibMLIRMIOpen, ${INCLUDE_DIR} is
-    # automatically propagated to the consumer library's include_dirs, which has the same
-    # effects as calling target_include_directories(MIOpen ${MLIRMIOpen_INCLUDE_DIR})
+    # so that when comsumer libraries (MIOpen) link in lib${LIBRARY_NAME}, ${INCLUDE_DIR} is
+    # automatically propagated to the consumer library's include_dirs
     install(TARGETS ${target_name}
       EXPORT ${export_set}
       ARCHIVE DESTINATION lib/${PACKAGE_NAME}
       COMPONENT ${component_name}
-      INCLUDES DESTINATION ${INCLUDE_DIR})
+      INCLUDES DESTINATION ${INCLUDE_DIR}
+    )
 
     # Generate package config and version file
     include(CMakePackageConfigHelpers)
     write_basic_package_version_file(
       "${CMAKE_CURRENT_BINARY_DIR}/${pkg_version_file}"
       VERSION ${CMAKE_PROJECT_VERSION}
-      COMPATIBILITY AnyNewerVersion)
+      COMPATIBILITY AnyNewerVersion
+    )
 
     configure_package_config_file(cmake/${pkg_config_file}.in
       "${CMAKE_CURRENT_BINARY_DIR}/${pkg_config_file}"
       INSTALL_DESTINATION lib/cmake/${PACKAGE_NAME}
-      PATH_VARS INCLUDE_DIR LIB_CMAKE_DIR)
+      PATH_VARS INCLUDE_DIR LIB_CMAKE_DIR
+      NO_SET_AND_CHECK_MACRO
+      NO_CHECK_REQUIRED_COMPONENTS_MACRO
+    )
 
     # Generate Targets.cmake file for the export-set to contain imported targets
-    set(ConfigPackageLocation lib/cmake/${PACKAGE_NAME})
     install(EXPORT ${export_set}
-      NAMESPACE Miir::
-      DESTINATION ${ConfigPackageLocation}
-      COMPONENT ${component_name})
+      NAMESPACE ${PACKAGE_NAME}::
+      DESTINATION ${LIB_CMAKE_DIR}
+      COMPONENT ${component_name}
+    )
 
     # Install package configuration and version files
     install(
       FILES
       "${CMAKE_CURRENT_BINARY_DIR}/${pkg_config_file}"
       "${CMAKE_CURRENT_BINARY_DIR}/${pkg_version_file}"
-      DESTINATION ${ConfigPackageLocation}
-      COMPONENT ${component_name})
+      DESTINATION ${LIB_CMAKE_DIR}
+      COMPONENT ${component_name}
+    )
   endfunction(export_target)
 
-  # Install libMLIRMIOpen.a as part of component libMLIRMIOpen
+  # Install lib${LIBRARY_NAME}.a as part of component lib${LIBRARY_NAME}
   # and export it to be searchable by find_package()
-  export_target(MLIRMIOpen libMLIRMIOpen)
+  export_target(${LIBRARY_NAME} lib${LIBRARY_NAME})
+
+  if ((NOT LLVM_INSTALL_TOOLCHAIN_ONLY) AND EXPORT_ALL_HEADERS)
+    install(DIRECTORY
+      ${PROJECT_SOURCE_DIR}/mlir/include/mlir
+      ${PROJECT_SOURCE_DIR}/mlir/include/mlir-c
+      DESTINATION include/${PACKAGE_NAME}
```

Comment:
Just to check, does this mean our header files will go to (supposing the prefix is `/usr`)
`/usr/include/rocMLIR/mlir/...`?

Reply:
Yes